daemon/endpoint.go

```diff
@@ -155,9 +155,7 @@ func (h *putEndpointID) Handle(params PutEndpointIDParams) middleware.Responder
 	add := labels.NewLabelsFromModel(params.Endpoint.Labels)
 
 	if len(add) > 0 {
-		endpointmanager.Mutex.Unlock()
-		errLabelsAdd := h.d.UpdateSecLabels(params.ID, add, labels.Labels{})
-		endpointmanager.Mutex.Lock()
+		errLabelsAdd := h.d.updateSecLabels(params.ID, add, labels.Labels{})
```

Comment:
Crap, you are right. I read this as Lock. The Unlock might be better after the Insert then. I'll close this.